management-service/services/message_queue_service.go

```diff
@@ -0,0 +1,53 @@
+package services
+
+import (
+	"fmt"
+
+	_pubsub "github.com/caraml-dev/xp/common/pubsub"
+	"github.com/caraml-dev/xp/common/segmenters"
+	"github.com/caraml-dev/xp/management-service/config"
+)
+
+type MessageQueueService interface {
```

Comment:
Two thoughts:
* This interface makes the existing `PubSubPublisherService` redundant. We can remove it and its mock under `mocks`.
* Shall we make a nested `messagequeue` package under `services` and and put `pubsub_message_queue.go` (which is largely `management-service/services/pubsub_publisher_service.go`) and `noop_message_queue.go` as the interface implementations for this service under that? That will make the code organization better.